src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,842 @@
+        /// </summary>
+        public uint Offset
+        {
+            get
+            {
+                return ReadRegisterValue(Register.Offset) & (uint)BitMask.Offset;
+            }
+
+            set
+            {
+                SetRegisterBits(Register.Offset, BitMask.Offset, value);
+            }
+        }
+
+        /// <summary>
+        /// The value of the full-scale calibration register
+        /// </summary>
+        public uint FullScale
+        {
+            get
+            {
+                return ReadRegisterValue(Register.FullScale) & (uint)BitMask.FullScale;
+            }
+
+            set
+            {
+                SetRegisterBits(Register.FullScale, BitMask.FullScale, value);
+            }
+        }
+
+        /// <summary>
+        /// Gets the metadata class for the device
+        /// </summary>
+        /// <returns></returns>
+        public static IDeviceMetadata GetDeviceMetadata()
+        {
+            return Ad7193.metadata;
+        }
+
+        /// <summary>
+        /// Initializes the ADC
+        /// </summary>
+        /// <param name="spiDevice">The SPI device to initialize the ADC on</param>
+        public Ad7193(SpiDevice spiDevice)
+        {
+            if ((spiDevice.ConnectionSettings.Mode & metadata.ValidSpiModes) != spiDevice.ConnectionSettings.Mode)
+            {
+                throw new Exception("SPI device must be in SPI mode 3 in order to work with AD7193.");
+            }
+
+            if (spiDevice.ConnectionSettings.ClockFrequency > metadata.MaximumSpiFrequency)
+            {
+                throw new Exception($"SPI device must have a lower clock frequency, because AD7193's maximum operating SPI frequncy is {metadata.MaximumSpiFrequency} Hz.");
+            }
+
+            _spiDevice = spiDevice;
+
+            Reset();
+        }
+
+        /// <summary>
+        /// Resets the ADC
+        /// </summary>
+        public void Reset()
+        {
+            for (int i = 0; i < 6; i++)
+            {
+                _spiDevice.Write(new byte[] { 0xFF });
+            }
+        }
+
+        /// <summary>
+        /// Initiates internal calibration, including zero-scale and full-scale calibrations
+        /// </summary>
+        public void Calibrate()
+        {
+            // internal zero scale calibration (MD2 = 1, MD1 = 0, MD0 = 0)
+            SetRegisterBits(Register.Mode, BitMask.ModeMD, 0b100);
+            WaitForADC();
+
+            // internal full scale calibration (MD2 = 1, MD1 = 0, MD0 = 1)
+            SetRegisterBits(Register.Mode, BitMask.ModeMD, 0b101);
+            WaitForADC();
+        }
+
+        /// <summary>
+        /// Waits after a single conversion until the DOUT/!RDY goes low to indicate the completion of a conversion
+        /// </summary>
+        public void WaitForADC()
+        {
+            while (!IsReady)
+            {
+                ReadRegisterValue(Register.Status);
+                Thread.Sleep(5);
```

Comment:
where is the sleep value coming from? Is the number documented in the spec? If so please mention the page